### PR TITLE
Cherrypick: Update cAdvisor godeps for v1.4.1

### PR DESCRIPTION
**Pull request by timstclair**
Fixes 2 issues:
- https://github.com/kubernetes/kubernetes/issues/32917
- https://github.com/kubernetes/kubernetes/issues/30069

Includes commits:
1. Moved google.golang.org/cloud/compute/metadata to cloud.google.com/go/compute/metadata - https://github.com/google/cadvisor/commit/159f002f9a9bfd3af56070f435a7f4287ff6b24b
2. Look at all cgroup mounts -https://github.com/google/cadvisor/commit/ac9388cf76723ef432e655a7f3ac6d67a926f64f
3. Update runc libcontainer dependency - https://github.com/google/cadvisor/commit/1e013abde8892d3ecc8b6aac6650af7ccf78c321

/cc @eparis @derekwaynecarr @pwittrock

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33806)

### CODE REVIEW

**Comment by timstclair** [comment]
Hmm, something's not right here. Will reassign once it's fixed.

**Comment by dchen1107** [comment]
Are we sure we want to cherrypick this to 1.4 branch?

**Comment by timstclair** [comment]
These were both fixes that were at some point targeted for v1.4, but we decided to hold for v1.4.1

File: pkg/kubelet/cm/helpers_linux.go
Comment (derekwaynecarr) [inline pkg/kubelet/cm/helpers_linux.go]: i think for containerized kubelet, we will also need this value to be true.

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]
we need to pass true in container manager.

File: pkg/kubelet/cm/helpers_linux.go
Comment (timstclair) [inline pkg/kubelet/cm/helpers_linux.go]: Done. Can you think of any possible adverse affects of this that I should test for?

Comment (derekwaynecarr) [inline pkg/kubelet/cm/helpers_linux.go]: no.  basically, the reason we needed to pass true in cadvisor is the same reason we need to pass true in kubelet.  we need to ensure we have the right mount points (from the host) and not local to the container.  this code is not really activated until kube 1.5 with the pod level cgroup work i am doing.

**Review by derekwaynecarr — APPROVED** [review]

**Comment by derekwaynecarr** [comment]
LGTM

**Comment by timstclair** [comment]
@dchen1107 Can you approve the cherry-pick request, or should I reassign to Phil?

**Comment by jessfraz** [comment]
@k8s-bot verify test this

**Comment by jessfraz** [comment]
@k8s-bot kubemark e2e test this

**Comment by jessfraz** [comment]
@timstclair is this only adding the 3 commits you mentioned to cadvisor or is it adding more? an 1,000 line diff into a point release is a little scary

**Comment by jessfraz** [comment]
ah I think the test failure is real

**Comment by timstclair** [comment]
> is this only adding the 3 commits you mentioned to cadvisor or is it adding more? an 1,000 line diff into a point release is a little scary

To clarify, those are the three commits from the cAdvisor godeps update, but we're transitively updating runc & gcloud deps. There are actually quite a few (124) commits in the runc update, so that is understandably concerning.

The total delta of runc is:
```
17 files changed, 124 insertions(+), 151 deletions(-)
```

We should probably switch to tracking a tagged runc version in the next release, otherwise we're going to run into this problem everytime a serious bug is found with runc.

The rest of the changes are in the new `cloud.google.com/go/...` dependency, which replaces `google.golang.org/cloud/...`. The diffs are small, but non-zero:
```
$ for f in $(find vendor/cloud.google.com/go/ -type f -name "*.go" | sed 's|vendor/cloud.google.com/go/||'); do echo -n "$f:  "; diff vendor/google.golang.org/cloud/$f vendor/cloud.google.com/go/$f | diffstat -s; done
compute/metadata/metadata.go:   1 file changed, 74 insertions(+), 18 deletions(-)
internal/cloud.go:   1 file changed, 5 insertions(+), 69 deletions(-)
```

> I think the test failure is real

When I rerun the test locally I get the opposite error:
```
hack/verify-godeps.sh 
Checking for 'Godeps/' changes against 'origin/master'
Starting to download all kubernetes godeps. This takes a while
# cd /tmp/gopath.1AXUwC/src/google.golang.org/cloud; git checkout eb47ba841d53d93506cfbfbc03927daf9cc48f88
godep: error restoring dep (google.golang.org/cloud/compute/metadata): chdir /tmp/gopath.1AXUwC/src/google.golang.org/cloud: no such file or directory
# cd /tmp/gopath.1AXUwC/src/google.golang.org/cloud; git checkout eb47ba841d53d93506cfbfbc03927daf9cc48f88
godep: error restoring dep (google.golang.org/cloud/internal): chdir /tmp/gopath.1AXUwC/src/google.golang.org/cloud: no such file or directory
godep: Error restoring some deps. Aborting check.
```

This error is what the cherrypick is working on fixing so I'll keep investigating

**Comment by timstclair** [comment]
In summary, I recognize the concerns, but I'm also not sure what the alternative is aside from cherrypicking directly into our vendored libs.

**Comment by jessfraz** [comment]
thanks for clarifying!

**Comment by timstclair** [comment]
@k8s-bot verify test this

**Comment by timstclair** [comment]
That latest verify failure makes more sense. It's the issue (https://github.com/kubernetes/kubernetes/issues/30069) this PR is partially fixing (the other half of the fix is in https://github.com/kubernetes/kubernetes/pull/32592)

**Comment by timstclair** [comment]
I can open a new PR which cherry-picks only the specific commits needed, based on the discussion in https://groups.google.com/d/topic/kubernetes-dev/GCvf664vVTg/discussion

What do we want to do about the cloud deps though? I suppose if we're not going to be using godep it doesn't matter that `google.golang.org/cloud` no longer exists.

**Comment by jessfraz** [comment]
I think thats ok :) and thanks so much, sorry this PR became a rabbit hole

**Comment by timstclair** [comment]
> I think thats ok

Sorry, which part were you referring to?

**Comment by jessfraz** [comment]
Both parts sgtm!

**Comment by jessfraz** [comment]
@timothysc what would be easiest is to ignore the google cloud stuff and have cadvisor cut a tag with just the patches we need, then update godep to use that

**Comment by timothysc** [comment]
s/timothysc/timstclair

**Comment by jessfraz** [comment]
ah so sorry!

**Comment by timstclair** [comment]
We can ignore the google cloud deps. We still need to decide what to do about runc though. Don't hold v1.4.1 for this, if it slips to 1.4.2 it's not a big deal.

**Comment by jessfraz** [comment]
I bet if we ask nicely they might make us a custom tag as well :)

**Comment by jessfraz** [comment]
can we pull the google cloud dep changes in here as well so the verification passes

**Comment by timstclair** [comment]
Do you mean https://github.com/kubernetes/kubernetes/pull/32592 ?

**Comment by jessfraz** [comment]
yes :)

**Comment by timstclair** [comment]
Done, but the cherrypick wasn't clean. I manually ignored the conflicting cAdvisor updates.

/cc @dims

**Comment by jessfraz** [comment]
thanks!

**Comment by jessfraz** [comment]
oh I know the problem w verify-godeps, i can fix it but ill wait for the other tests to pass and merge this one by hand when they are green

**Comment by timstclair** [comment]
Thanks!

File: vendor/github.com/google/cadvisor/metrics/prometheus.go
Comment (soltysh) [inline vendor/github.com/google/cadvisor/metrics/prometheus.go]: How this got in, it's not part of the linked cadvisor version, @derekwaynecarr @smarterclayton?

**Pull request merged**
